src/components/WeatherResults.js

```diff
@@ -1,65 +1,51 @@
 /** @jsx createElement */
 /** @jsxFrag createFragment */
-import { createElement, createFragment } from '../framework/element';
-import renderApp from '../framework/render';
+import { createElement, createFragment, useState } from '../framework';
+import { CELSIUS_UNITS } from '../utils';
 
 import UnitSwitch from './UnitSwitch';
 import WeatherToday from './WeatherToday';
 import WeatherForecast from './WeatherForecast';
-import {
-  getCurrentCityWeatherData,
-  getFilteredByDateWeatherData,
-  isCurrentCityDataLoaded,
-} from '../data/weatherData';
+import { getFilteredByDateWeatherData } from '../data/weatherData';
 
-function setCurrentUnits(value) {
-  window.dataStore.currentUnits = value;
-  renderApp();
-}
-
-function WeatherResults() {
-  const { isDataLoading, currentUnits, error, currentCity } = window.dataStore;
-  let content = null;
+function WeatherResults({ isLoading, error, currentCity, weatherData: cityByWeather }) {
+  const [currentUnits, setCurrentUnits] = useState(CELSIUS_UNITS);
 
-  if (currentCity === '') {
-    content = 'Search by city name';
-  } else {
-    if (error !== null) {
-      content = error;
-    }
+  if (!currentCity) {
+    return <div>Search by city name</div>;
+  }
 
-    if (isDataLoading) {
-      content = 'Loading...';
-    }
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
 
-    if (isCurrentCityDataLoaded()) {
-      const weatherData = getCurrentCityWeatherData();
-      const weatherTodayData = getFilteredByDateWeatherData(weatherData, { includeBaseDate: true });
-      const weatherForecastData = getFilteredByDateWeatherData(weatherData, {
-        includeDatesAfterBase: true,
-      });
-      content = (
-        <>
-          <UnitSwitch currentUnits={currentUnits} setCurrentUnitsCB={setCurrentUnits} />
-          <br />
-          <WeatherToday
-            currentUnits={currentUnits}
-            currentCity={currentCity}
-            weatherData={weatherTodayData}
-          />
-          <br />
-          <br />
-          <WeatherForecast
-            currentUnits={currentUnits}
-            currentCity={currentCity}
-            weatherData={weatherForecastData}
-          />
-        </>
-      );
-    }
+  if (error) {
+    return <div>{error}</div>;
   }
 
-  return <p>{content}</p>;
+  const weatherData = cityByWeather?.list;
```

Comment:
Does it build's for you correctly?
Parcel bundler fails for me previously because of `?.` operator